rustls/src/msgs/deframer.rs

```diff
@@ -855,8 +855,13 @@ mod tests {
             record_layer: &mut RecordLayer,
             negotiated_version: Option<ProtocolVersion>,
         ) -> Result<Option<Deframed>, Error> {
-            self.inner
-                .pop(record_layer, negotiated_version, &mut self.buffer)
+            let mut deframer_buffer = self.buffer.borrow();
```

Comment:
Same here with the discard vs error returns.

Reply:
same answer as the previous one